### ISSUE TITLE
What type of server is required?

### ISSUE THREAD

**Issue by Russ76** [trzy/MRLaserTag#1]
For using one headset and one camera, what are the specs of CPU and RAM necessary for acceptable performance?

**Comment by trzy** [comment]
I don’t actually know what the minimum spec is but I suspect as long as your system has the two USB-C ports needed to run the cameras, virtually any system should be capable.

Note that this prototype does not perform continuous tracking or object detection. When it starts up, there’s a 10 second countdown and at the end of that, it saves the environment state (walls and objects) and sends that to any headsets and spectators that subsequently connect. During that initial 10 second period, AprilTag detection runs continuously, which *is* CPU intensive (if you run mrlasertag with —show-fps, you’ll see the frame rate drop considerably during that time) but no averaging over time is being performed and only the very last detected frame is used. The countdown was put in place to give me a chance to get out of the way in time in case I was obstructing any AprilTags while sitting in front of the camera.

So to summarize: because the wall and object detection are run only once, relying on the headset’s SLAM to do the heavy lifting of tracking itself in the room after that, very little work happens on the server. 

You should be able to build and run the server process pretty easily even without all the other components. It’ll pop up one window per camera (up to two cameras supported) plus a depth window from the main camera and you can observe its performance.

**Comment by Russ76** [comment]
Sounds good! My headset is coming this week! The Oak-D camera soon. I already own Zed and Realsense D435 stereo cameras. I'll try to use one of the latter with the Oak-D to have two running. May have to pair the Nvidia TX1 with one camera for processing!
